### PR TITLE
[WIP] xarray container

### PR DESCRIPTION
**Pull request by mmccarty**

### CODE REVIEW

File: intake/container/__init__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from .semistructured import SemiStructured
+
+container_map = {
+    'dataframe' : DataFrame,
+    'python' : SemiStructured,
+    'ndarray' : NdArray
+}
+
+def get_container_klass(container):
+    return container_map.get(container, BaseContainer)
```
Comment (mmccarty) [inline intake/container/__init__.py]: Using a mapper here for the time being so we don't have to change the interface.

File: intake/container/__init__.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from .base import BaseContainer
+from .dataframe import DataFrame
+from .ndarray import NdArray
+from .semistructured import SemiStructured
+
+container_map = {
+    'dataframe' : DataFrame,
+    'python' : SemiStructured,
```
Comment (martindurant) [inline intake/container/__init__.py]: This name is misleading: it is a sequence of arbitrary python objects; in the common case those objects are homogeneous, probably dicts, but that is not a requirement.

File: intake/container/__init__.py (hunk 1 above)
Comment (martindurant) [inline intake/container/__init__.py]: Totally fine with that. Later we may provide a "register" API, function or decorator.

File: intake/container/__init__.py (hunk 2 above)
Comment (mmccarty) [reply intake/container/__init__.py]: Yeah, I agree.  I struggled with the name here.  I certainly didn't want to use `Python`.  SemiStructure came from the docs.  In either case, the naming should be consistent.

**Comment by mmccarty** [comment]
:+1: for integration test repo

**Pull request closed without merging**